### PR TITLE
OSAC-1702: Add multi-tenant resource isolation E2E tests (Phase 4)

### PR DESCRIPTION
**Pull request by obochan-rh**
## Summary

- Adds 3 new E2E integration test specs for multi-tenant resource isolation (Phase 4 of Auth/Authz testing)
- Tests verify that resources created by one tenant are invisible to other tenants via List and Get, that cross-tenant deletion is rejected, and that each tenant's List returns only their own resources
- Uses VirtualNetwork as the test resource with break-glass user authentication per tenant

## Test Specs

| Spec | Validates |
|------|-----------|
| Resources created by tenant-A are not visible to tenant-B | Cross-tenant List filtering + Get returns NotFound |
| Tenant-B cannot delete tenant-A's resources | Cross-tenant Delete returns NotFound, resource persists |
| Each tenant sees only their own resources in List | Bidirectional isolation with concurrent resources |

## Test Results

- **Main branch (baseline):** 146/146 passed
- **Feature branch:** 149/149 passed (146 existing + 3 new)

## Test Plan

- [x] Phase 4 tests pass in isolation (`--focus "Multi-tenant resource isolation"`)
- [x] Full test suite passes on main (146/146)
- [x] Full test suite passes on feature branch (149/149)
- [x] No regressions introduced

Resolves: [OSAC-1702](https://redhat.atlassian.net/browse/OSAC-1702)

## Summary by CodeRabbit

* **New Features**
  * Added end-to-end coverage for multi-tenant resource isolation.
  * Verified that network resources created in one tenant are only visible to that tenant, and cannot be listed, fetched, or deleted by another tenant.
  * Included cleanup handling for test-created resources and connections.

### CODE REVIEW

**Review by jhernand — APPROVED** [review]

File: it/it_tenant_lifecycle_test.go [hunk 1]
```diff
@@ -702,3 +702,244 @@ var _ = Describe("Tenant edge cases and resilience", func() {
...
+		Expect(getResp.GetObject().GetId()).To(Equal(vnId))
+
+		By("Verifying tenant-B cannot see tenant-A's VirtualNetwork via List")
+		vnClientB := publicv1.NewVirtualNetworksClient(extConnB)
+		listResp, err := vnClientB.List(ctx, publicv1.VirtualNetworksListRequest_builder{}.Build())
+		Expect(err).ToNot(HaveOccurred())
+		for _, item := range listResp.GetItems() {
+			Expect(item.GetId()).ToNot(Equal(vnId),
+				"tenant-B should not see tenant-A's VirtualNetwork in list")
+		}
```
[the remark this answers is not in the document]
Comment (obochan-rh) [reply it/it_tenant_lifecycle_test.go:796]: Fixed — added explicit `Limit: 1000` to all three List calls so the assertions aren't affected by the default DAO page size of 100. Updated in the latest force-push.

**Comment by obochan-rh** [comment]
@coderabbitai resolve

**Comment by obochan-rh** [comment]
@jhernand The `lgtm` label was removed after the force-push that addressed CodeRabbit's review comment (added explicit List limits). Could you re-`/lgtm` when you get a chance? All CI checks are green. Thanks!

**Review by jhernand — APPROVED** [review]
